private/bufpkg/bufconfig/bufconfig.go

```diff
@@ -143,6 +143,35 @@ func WriteConfigWithDependencyModuleReferences(dependencyModuleReferences ...buf
 	}
 }
 
+// WriteConfigWithBreakingConfig returns a new WriteConfigOption that sets the breaking change
+// config for the module.
+//
+// If this option is used and the version used will be consistent with the rest of the config.
+func WriteConfigWithBreakingConfig(breakingConfig *bufbreakingconfig.Config) WriteConfigOption {
+	return func(writeConfigOptions *writeConfigOptions) {
+		writeConfigOptions.breakingConfig = breakingConfig
+	}
+}
+
+// WriteConfigWithLintConfig returns a new WriteConfigOption that sets the lint config for // the module.
+//
+// If this option is used and the version used will be consistent with the rest of the config.
+func WriteConfigWithLintConfig(lintConfig *buflintconfig.Config) WriteConfigOption {
+	return func(writeConfigOptions *writeConfigOptions) {
+		writeConfigOptions.lintConfig = lintConfig
+	}
+}
+
+// WriteConfigWithVersion returns a new WriteConfigOption that sets the version of the config
+// being written.
+//
+// If this is not set, the default is v1.
```

Comment:
Depending on the conversation below, the default might be `v1beta1` to preserve the default value of `buf.yaml`, but I'm not totally sure yet if this option will match that behavior from a `WriteConfig` standpoint.